File: Data Structures & Algorithms/eating-bananas/submission-6.js
class Solution {
    /**
     * @param {number[]} piles
     * @param {number} h
     * @return {number}
     */
    minEatingSpeed(piles, h) { 

        let l = 1 
        let r = Math.max(...piles)  
        let res = r

        while(l <= r) {
            let mid = Math.floor( ( l + r ) / 2 ) 
            let total = 0 

            for (const pile of piles) {
                let hours = Math.ceil( pile / mid ) 
                total += hours  
            }

            if (total <= h) {
                res = mid 
                r = mid - 1
            } else {
                l = mid + 1
            } 
        } 

        return res
    }
}

File: Data Structures & Algorithms/maximum-subarray/submission-6.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxEnding= nums[0]
        res = nums[0] 

        for i in range(1, len(nums)): 
            maxEnding = max(maxEnding + nums[i], nums[i]) 
            res = max(res, maxEnding) 

        return res